airflow/metrics/datadog_logger.py

```diff
@@ -66,109 +96,95 @@ def __init__(
         self.metrics_tags = metrics_tags
         self.metric_tags_validator = metric_tags_validator
 
+    @prepare_stat_with_tags
     @validate_stat
     def incr(
         self,
-        stat: str,
+        metric_name: str,
         count: int = 1,
         rate: float = 1,
         *,
-        tags: dict[str, str] | None = None,
+        tags: list[str] | None = None,
```

Comment:
I see what's going on here (the decorator changes the dict into the list) but I wonder if there's an easier way of dealing with this... 🤔

Reply:
Since Datadog expecting a list I don't think if we can do better here. Since the decorator would be called first, it does not break any test cases. I can revert it back, but there would be many redundant codes for each of the functions.

Reply:
I'm more worried about type completion, but I guess if nothing is complaining this is fine.